provider/resources.go

```diff
@@ -2868,6 +2868,20 @@ func Provider() *tfbridge.ProviderInfo {
 					}
 					return config, nil
 				},
+				TransformFromState: func(ctx context.Context, inputs resource.PropertyMap) (resource.PropertyMap, error) {
+					if _, ok := inputs["db_name"]; ok {
+						return inputs, nil
+					}
+					name, ok := inputs["name"]
+					if !ok {
+						// It's seems like inputs are not in a valid state
+						return inputs, nil
+					}
+					tfbridge.GetLogger(ctx).Debug(`migrating from "name" to "dbName"`)
+					inputs["dbName"] = name
+					delete(inputs, "name")
```

Comment:
I'm curious that this helps out the diff but it doesn't actually migrate right. It doesn't rewrite the inputs in the state. Perhaps Check does that already?

Basically my question is whether after all is said and done and users upgrade successfully, they still have "name" in their state, or the normal pathway completely cleans it out now.

Reply:
It depends. Pulumi doesn't have a mechanism to persist to the state beyond the normal CRUD lifecycle. If the user runs an update (preview=false) or refresh (preview=false), then their new state will be corrected. If they just run `pulumi preview` then this hook will run each time but never correct their state.

Reply:
But we're asking the users to correct the program inputs in Check with a warning right. So I think once they correct program inputs and do a pulumi up, I wonder if we update to new state or it's a no-op diff. Eh. This is tricky to reason about - I worry that the old states are going to hang around indefinitely, possibly; there's no easy way for Pulumi providers to issue state corrections that migrate Pulumi states with Updates or some such but otherwise leave things as-is.

Reply:
I believe that they will stay on the old state until they do something else to the resource. To resolve the underlying issue, either the pulumi CLI or the bridge would need to support a "migration" concept, where a resource declares that it needs to update state because of a new provider version instead of a user change.